src/basic/mkdir.c

```diff
@@ -248,16 +249,27 @@ int mkdir_p_root(const char *root, const char *p, uid_t uid, gid_t gid, mode_t m
                 return r;
         }
 
-        if (uid_is_valid(uid) || gid_is_valid(gid)) {
-                _cleanup_close_ int nfd = -EBADF;
+        if (ts == USEC_INFINITY && !uid_is_valid(uid) && !gid_is_valid(gid))
```

Comment:
`!timestamp_is_set(ts)` ? I mean, we should also ignore `ts == 0`. Or does this intentionally accept `ts == 0`?

Reply:
But setting 0 timestamp sounds spurious for me... Or `futimens()` handles 0 gracefully?

Reply:
Yes, `futimens()` handles it. It sets the timestamps to 0 like so: 

```
  File: /var/tmp/.#repartc9a185024a9ea6f2/somedir
  Size: 4096      	Blocks: 8          IO Block: 4096   directory
Device: fd00h/64768d	Inode: 6686698     Links: 2
Access: (0755/drwxr-xr-x)  Uid: ( 1001/      ms)   Gid: ( 1001/      ms)
Access: 1970-01-01 00:00:00.000000000 +0000
Modify: 1970-01-01 00:00:00.000000000 +0000
```

Reply:
OK. Still I want to filter out 0, but not a strong opinion. Feel free to ignore this.